agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrUploadService.java

```diff
@@ -58,16 +59,22 @@ public void receive(AlertBreach alertBreach, long timestamp, File file) {
 
     jfrUploader
         .uploadJfrFile(
+            UUID.fromString(alertBreach.getProfileId()),
```

Comment:
`UUID` seems generated, converted into a String and then, at this point (JFR file uploading) the `String` is reconverted into `UUID`. I may miss something. `UUID` could not be always used and propagated until the JFR loading?

Reply:
AlertBreach does get serialized and read from a number of places. Unfortunately putting a UUID into it would then necessitate adding Jackson libraries to `agent-alerting-api` for it to understand how to serialize/deserialize a UUID. Given the frequent CVEs that you get from Jackson I would prefer not to add it as a dependency for such a small use case. I think the alternative would be to just treat the id from here on out as a string rather than a UUID, however the reality of the api is that it is required to be a UUID.